src/clients/python/experimental_api_v2/library/grpcclient.py

```diff
@@ -0,0 +1,674 @@
+
+        Returns
+        -------
+        InferResult
+            The object holding the result of the inference, including the
+            statistics.
+
+        Raises
+        ------
+        InferenceServerException
+            If server fails to issue inference.
+        """
+
+        def wrapped_callback(call_future):
+            try:
+                result = InferResult(call_future.result())
+            except grpc.RpcError as rpc_error:
+                raise_error_grpc(rpc_error)
+            callback(result=result)
+
+        self._request = grpc_service_v2_pb2.ModelInferRequest()
+        self._request.model_name = model_name
+        self._request.model_version = model_version
+        self._request.sequence_id = sequence_id
+        for infer_input in inputs:
+            self._request.inputs.extend([infer_input._get_tensor()])
+        for infer_output in outputs:
+            self._request.outputs.extend([infer_output._get_tensor()])
+
+        try:
+            self._call_future = self._client_stub.ModelInfer.future(
+                self._request)
+            self._call_future.add_done_callback(wrapped_callback)
+        except grpc.RpcError as rpc_error:
+            raise_error_grpc(rpc_error)
+
+
+class InferInput:
+    """An object of InferInput class is used to describe
+    input tensor for an inference request.
+
+    Parameters
+    ----------
+    name : str
+        The name of input whose data will be described by this object
+
+    """
+
+    def __init__(self, name):
+        self._input = grpc_service_v2_pb2.ModelInferRequest().InferInputTensor()
+        self._input.name = name
+
+    @property
+    def name(self):
+        """Get the name of input associated with this object.
+
+        Returns
+        -------
+        str
+            The name of input
+        """
+        return self._input.name
+
+    @property
+    def datatype(self):
+        """Get the datatype of input associated with this object.
+
+        Returns
+        -------
+        str
+            The datatype of input
+        """
+        return self._input.datatype
+
+    @property
+    def shape(self):
+        """Get the shape of input associated with this object.
+
+        Returns
+        -------
+        str
+            The shape of input
+        """
+        return self._input.shape
+
+    def set_data_from_numpy(self, numpy_array):
+        """Set the tensor data (datatype, shape, contents) for
+        input associated with this object.
+
+        Parameters
+        ----------
+        numpy_array : numpy array
+            The tensor data in numpy array format
+        """
+        self._input.datatype = np_to_trtis_dtype(numpy_array.dtype)
+        self._input.shape.extend(numpy_array.shape)
+        self._input_contents = grpc_service_v2_pb2.InferTensorContents()
+        self._input_contents.raw_contents = numpy_array.tobytes()
+        self._input.contents.CopyFrom(self._input_contents)
```

Comment:
This copy can be very expensive... can't we just assign directly to input.contents.